src/pudl/transform/eiaaeo.py

```diff
@@ -0,0 +1,324 @@
+    ).drop(
+        columns=[
+            "series_name",
+            "category_name",
+            "variable_fields",
+        ]
+    )
+
+    sanitized = enriched.assign(
+        **{
+            colname: __sanitize_string(enriched[colname])
+            for colname in enriched.columns
+            if colname not in {"projection_year", "value"}
+        }
+    ).pipe(
+        lambda df: df.assign(
+            report_year_series=df.report_year_series.str.replace("aeo", ""),
+            value=pd.to_numeric(df.value.str.replace("- -", "")),
+        )
+    )
+
+    assert (
+        sanitized.region_series == sanitized.region_category
+    ).all(), "Series and category must agree on region!"
+    assert (
+        sanitized.case_series == sanitized.model_case_eiaaeo
+    ).all(), "Series and taxonomy must agree on case!"
+
+    # TODO 2024-04-20: one day we'll include report year in the extract step -
+    # at that point we'll have to compare that with the report year extracted
+    # from the series name.
+    return sanitized.drop(columns=["region_series", "case_series"]).rename(
+        columns={"region_category": "region", "report_year_series": "report_year"}
+    )
+
+
+def unstack(df: pd.DataFrame, eventual_pk: list[str]):
+    """Unstack the values by the various variable names provided.
+
+    2024-04-27: NB it's important that `projection_year` is the last level of
+    index, so that when we sort and diff() we are actually subtracting year
+    over year.
+    """
+    assert (
+        eventual_pk[-1] == "projection_year"
+    ), "Projection year must be last level of index!"
+    unstacked = (
+        df.set_index(eventual_pk + ["variable_name"])
+        .unstack(level="variable_name")
+        .sort_index()
+    )
+    unstacked.columns = unstacked.columns.droplevel()
+    unstacked.columns.name = None
+    return unstacked
+
+
+@asset(io_manager_key="pudl_io_manager")
+def core_eiaaeo__yearly_projected_generation_in_electric_sector_by_technology(
+    raw_eiaaeo__electric_power_projections_regional,
+):
+    """Projected net summer generation capacity and additions/retirements."""
+
+    def _drop_aggregated_generation_data(df: pd.DataFrame) -> pd.DataFrame:
+        """These "aggregate" data series aren't split out by technology.
+
+        We'll report them elsewhere - but for this resource we filter them out.
+        """
+        generation_mask = df.variable_name == "generation"
+        is_aggregated_mask = df.dimension.isin(
+            {
+                "total_generation",
+                "sales_to_customers",
+                "generation_for_own_use",
+            }
+        )
+        return df.loc[~(generation_mask & is_aggregated_mask)]
+
+    def _collect_totals(df: pd.DataFrame) -> pd.DataFrame:
+        """Various total columns are named differently - they don't have to be."""
+        return df.assign(
+            dimension=df.dimension.str.replace(r"^total.*$", "total", regex=True)
+        )
```

Comment:
wait are the columns named differently or the values in this dimension column named differently?

Reply:
The values in the dimension column are named differently *and* the columns are named differently. So we had "cumulative_unplanned_additions|total_additions" and then "cumulative_planned_additions|total_planned_additions". Which was very silly since those `total_*` dimensions would just have null values for the non-applicable columns. Hence combining them into just `total`.